shotover-proxy/src/frame/cassandra.rs

```diff
@@ -483,40 +457,134 @@ impl CassandraOperation {
+    pub fn get_table_name(statement: &CassandraStatement) -> Option<&FQName> {
+        match statement {
+            CassandraStatement::AlterTable(t) => Some(&t.name),
+            CassandraStatement::CreateIndex(i) => Some(&i.table),
+            CassandraStatement::CreateMaterializedView(m) => Some(&m.table),
+            CassandraStatement::CreateTable(t) => Some(&t.name),
+            CassandraStatement::Delete(d) => Some(&d.table_name),
+            CassandraStatement::DropTable(t) => Some(&t.name),
+            CassandraStatement::DropTrigger(t) => Some(&t.table),
+            CassandraStatement::Insert(i) => Some(&i.table_name),
+            CassandraStatement::Select(s) => Some(&s.table_name),
+            CassandraStatement::Truncate(t) => Some(t),
+            CassandraStatement::Update(u) => Some(&u.table_name),
+            _ => None,
         }
     }
+}
 
-    pub fn parse_from_string(sql: String) -> Self {
-        match Parser::parse_sql(&GenericDialect, &sql) {
-            _ if sql.contains("ALTER TABLE") || sql.contains("CREATE TABLE") => {
-                tracing::error!("Failed to parse CQL for frame {:?}\nError: Blacklisted query as sqlparser crate cant round trip it", sql);
-                CQL::FailedToParse(sql)
-            }
-            Ok(ast) if ast.is_empty() => {
-                tracing::error!(
-                    "Failed to parse CQL for frame {:?}\nResulted in no statements.",
-                    sql
-                );
-                CQL::FailedToParse(sql)
-            }
-            Ok(mut ast) => CQL::Parsed(Box::new(ast.remove(0))),
-            Err(err) => {
-                tracing::error!("Failed to parse CQL for frame {:?}\nError: {:?}", sql, err);
-                CQL::FailedToParse(sql)
-            }
+pub fn parse_statement(cql_query_str: &str) -> CassandraStatement {
+    let mut ast = CassandraAST::new(cql_query_str);
+
+    // TODO: implement batch support, currently batch messages are discarded as Unknown
+    if ast.has_error() || ast.statements.len() != 1 {
+        CassandraStatement::Unknown(cql_query_str.to_string())
+    } else {
+        ast.statements.remove(0).statement
+    }
+}
+
+fn from_string_value(value: &str) -> CassandraType {
+    if value.starts_with('\'') || value.starts_with("$$") {
+        CassandraType::Varchar(Operand::unescape(value))
+    } else if value.starts_with("0X") || value.starts_with("0x") {
+        hex::decode(&value[2..])
+            .map(|x| CassandraType::Blob(Blob::from(x)))
+            .unwrap_or(CassandraType::Null)
+    } else if let Ok(n) = i64::from_str(value) {
+        CassandraType::Bigint(n)
+    } else if let Ok(n) = f64::from_str(value) {
+        CassandraType::Double(n)
+    } else if let Ok(uuid) = Uuid::parse_str(value) {
+        CassandraType::Uuid(uuid)
+    } else if let Ok(ipaddr) = IpAddr::from_str(value) {
+        CassandraType::Inet(ipaddr)
+    } else {
+        CassandraType::Null
+    }
```

Comment:
Could we write this as: 
```rust
    if value.starts_with('\'') || value.starts_with("$$") {
        CassandraType::Varchar(Operand::unescape(value))
    } else if value.starts_with("0X") || value.starts_with("0x") {
        hex::decode(&value[2..])
            .map(|x| CassandraType::Blob(Blob::from(x)))
            .unwrap_or(CassandraType::Null)
    }
   
    if let Ok(n) = i64::from_str(value) {
        CassandraType::Bigint(n)
    } else if let Ok(n) = f64::from_str(value) {
        CassandraType::Double(n)
    } else if let Ok(uuid) = Uuid::parse_str(value) {
        CassandraType::Uuid(uuid)
    } else if let Ok(ipaddr) = IpAddr::from_str(value) {
        CassandraType::Inet(ipaddr)
    } else {
        CassandraType::Null
    }
```

I think it more clearly illustrates that we must check the first 2 cases first and the order of the others doesn't matter.

Reply:
I dont see why the order of the others doesnt matter.
I'm pretty sure swapping f64::from_str and i64::from_str would break things.

Reply:
True, my mistake.